src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEdit.tsx

```diff
@@ -85,169 +94,216 @@ const GanttTaskBarEdit = ({
+                  <Typography
+                    variant="body1"
+                    sx={{
+                      color: task.styles ? task.styles.color : '#ffffff',
+                      px: 1,
+                      overflow: 'hidden',
+                      textOverflow: 'ellipsis',
+                      whiteSpace: 'nowrap'
+                    }}
+                  >
+                    {task.name}
+                  </Typography>
+                </Box>
               </Box>
-            </Box>
 
-            <Box
-              sx={{
-                cursor: isProject ? 'default' : 'ew-resize',
-                height: '100%',
-                width: '5rem',
-                position: 'relative',
-                right: '-10'
+              <Box
+                sx={{
+                  cursor: isProject ? 'default' : 'ew-resize',
+                  height: '100%',
+                  width: '5rem',
+                  position: 'relative',
+                  right: '-10'
+                }}
+                onMouseDown={isProject ? undefined : handleMouseDown}
+              />
+            </Box>
+          </div>
+          {isProject && (
+            <Chip
+              label={'+'}
+              onClick={() => {
+                setShowAddWorkPackageModal(true);
               }}
-              onMouseDown={isProject ? undefined : handleMouseDown}
             />
-          </Box>
-        </div>
-        {isProject && (
-          <Chip
-            label={'+'}
-            onClick={() => {
-              setShowAddWorkPackageModal(true);
-            }}
+          )}
+        </Box>
+      </div>
+      {task.unblockedWorkPackages.map((workPackage) => {
+        return (
+          <GanttTaskBarEdit
+            key={workPackage.id}
+            days={days}
+            task={transformWorkPackageToGanttTask(workPackage, task.teamName, task.totalWorkPackages)}
+            createChange={createChangeHandler}
+            getStartCol={getStartCol}
+            getEndCol={getEndCol}
+            isProject={false}
+            addWorkPackage={addWorkPackage}
+            getNewWorkPackageNumber={getNewWorkPackageNumber}
+          />
+        );
+      })}
+      {task.blocking.map((wbsNum) => {
+        const workPackage = task.totalWorkPackages.find((wp) => wbsPipe(wp.wbsNum) === wbsPipe(wbsNum));
+        if (!workPackage) return <></>;
+
+        return (
+          <GanttTaskBarBlockedEdit
+            key={wbsPipe(wbsNum)}
+            days={days}
+            task={transformWorkPackageToGanttTask(workPackage, task.teamName, task.totalWorkPackages)}
+            createChange={createChangeHandler}
+            getStartCol={getStartCol}
+            getEndCol={getEndCol}
+            addWorkPackage={addWorkPackage}
+            getNewWorkPackageNumber={getNewWorkPackageNumber}
           />
-        )}
-      </Box>
-    </div>
+        );
+      })}
+    </>
```

Comment:
this component in particular is pretty heinous to read through (this is definitely a better place to have the complexity concentrated though). Where you can, break sections of this that make sense into helper components with good names (same for the other GanttTaskBar components where it makes sense)